enclave-persistence/README.md

```diff
@@ -0,0 +1,83 @@
+## Enclave persistence
+
+This is a sample to demonstrate the use of persistent map in your enclave. Conclave provides two ways to persist
+information: the persistent map and persistent filesystem. The persistent filesystem can be used to persist files and
+make them available even during enclave restarts. On the other hand, the persistent map is a persistent, encrypted
+key-value store that can be used to store information from your enclave. The enclave can use this key-value store to
+persist information as required.Both persistent map and persistent filesystem come with their pros and cons. Which one
+to use depends on the application logic and how much the users of the enclave trust the host.
+
+1. Rollback attacks:The persistent filesystem is susceptible to rewind(or rollback) attacks.A malicious host can perform
+   a rewind (or rolllback) attack by deliberately restarting the enclave, providing an older image of the encrypted
+   filesystem.On the other hand, persistent map is resistant to it.
+2. Transactionality: Writes to the persistent filesystem are made immediately, i.e. they are not transactional. This
+   means they are fast and are not buffered. However, it also means changes are still persisted if the enclave
+   subsequently throws an exception. This has implications if the state of the enclave needs to be in sync with the
+   state of its clients. If the changes that were persisted are linked to mail that was supposed to be sent to clients (
+   but was not due to the exception) then the state of the clients will not be in-line. In case of persistent map
+   changes to the persistent map are only committed at the end of every `receiveMail` and
+   `receiveFromUntrustedHost` call where its entire contents is serialized and encrypted into a single blob. This
+   enables transactionality with respect to the processing of mail.
+3. However, one must note that persistent map does come at the cost of a performance overhead and so the map should be
+   used judiciously. More details about the persistent map and persistent filesystem could be
+   found [here](https://docs.conclave.net/persistence.html#conclave-filesystems)
+
+This Conclave sample demonstrates password authentication within an enclave. The enclave performs 2 major tasks:
+
+1. Stores the hash of the password in Persistent Map
+2. Validates a given password with the stored hash. Additional details: SHA-512 is being used to perform hash
+   calculation. The username is used as salt. For stronger security, using SecureRandom class for salt generation is
+   recommended.
```

Comment:
Realise that the use of salts in passwords for existing services is to guard against password leaks if the user table is leaked or someone gets access to the the hashes. However, since the passwords in our sample are encrypted using the enclave's sealing key, even if someone hacks into the server and gets access to the encrypted data, they can't read it. if they could read it it actually implies that they've cracked the enclave open, which is a much bigger issue! So I would argue the use of the salt or password hashing is not at all needed.

Reply:
WDYT @roy-hopkins?